config/v1/types_infrastructure.go

```diff
@@ -1188,6 +1208,19 @@ type VSpherePlatformFailureDomainSpec struct {
 	// +kubebuilder:validation:Required
 	Zone string `json:"zone"`
 
+	// regionAffinity holds the type of region
+	// +openshift:validation:featureGate=VSphereHostVMGroupZonal
+	// +kubebuilder:validation:Optional
+	RegionAffinity VSphereFailureDomainRegionAffinity `json:"regionAffinity,omitempty"`
```

Comment:
As this has `omitempty`, it needs to be a pointer else the json marshaller won't honour the omitempty

We only need to omit structs in configuration APIs when they have required fields, which I believe this one does

Reply:
ack